Refactor and extend palette parsing script

JasonJohansson opened the pull request:
Modularized parse_palette.py with functions for parsing, formatting, and saving palette data. Added CLI options for auto-inserting map palettes into forge.py, saving to a file, and improved error handling and user guidance. The script now supports easier integration and automation for map palette management.

JasonJohansson committed:
Refactor and extend palette parsing script

Modularized parse_palette.py with functions for parsing, formatting, and saving palette data. Added CLI options for auto-inserting map palettes into forge.py, saving to a file, and improved error handling and user guidance. The script now supports easier integration and automation for map palette management.

Waffle1434 commented on Addon/parse_palette.py:
```diff
@@ -1,21 +1,244 @@
+        # Create backup
+        backup_path = forge_py_path.with_suffix('.py.backup')
+        with open(backup_path, 'w', encoding='utf-8') as f:
+            f.write(content)
```
No need, forge.py is tracked in git

Waffle1434 commented on Addon/parse_palette.py:
```diff
@@ -1,21 +1,244 @@
+    result = "TwoWayDict({"
+    result += ", ".join([f"{k}: '{v}'" for k, v in items])
+    result += "})"
+    return result
```
No need for a function.

JasonJohansson committed:
Refactor parse_palette.py for conciseness

Simplified docstrings, print statements, and file operations for improved readability and maintainability. Replaced verbose comments and multi-line strings with concise one-liners, and streamlined output formatting throughout the script.

JasonJohansson committed:
fix tabs and newlines
